src/client/components/equalizer/equalizer.js

```diff
@@ -0,0 +1,174 @@
+import React from 'react';
+import Hark from 'hark';
+import PropTypes from 'prop-types';
+import Graphicequaliser from './canvasequalizer/canvasequalizer';
+import PlayButton from './playbutton/playbutton';
+import Uploadbutton from './upload/uploadbutton';
+import Infoabouttrack from './upload/infoaboutfile/infoaboutuploadfile';
+import Streambutton from './streambutton/streambutton';
+import { connect } from 'react-redux';
+import { createAudioData, playPauseSoundFromFile, createStreamData, startMuteStreamAudio, mergeCanvasWidth } from '../../actions/audioactions';
+
+
+class Equalizer extends React.Component {  
+   componentDidMount(){ 
+      this.detectStreamSoundFromMicrophon();
+    }
+   
+  detectStreamSoundFromMicrophon=()=>{    
+    const context = this.props.audioData.audiocontext;          
+    const audioLineIn = new Audio();   
+    if (navigator.mediaDevices) {      
```

Comment:
in such cases you can avoid nesting using inverted check, e.g.

```
if (!navigator.mediaDevices) { throw new Error('browser doesnt support audio API'); }
navigator.mediaDevices.getUserMedia ({audio: true})      
      .then(...)
      .catch()
```
also please move logic from  then to separate method, so you will have more compact and readable code, e.g.

```
.then(createStream)
.catch(handleError)
```

Reply:
done